tesseract/tesseract_planning/tesseract_command_language/src/utils/utils.cpp

```diff
@@ -48,6 +48,52 @@ const Eigen::VectorXd& getJointPosition(const Waypoint& waypoint)
   throw std::runtime_error("Unsupported waypoint type.");
 }
 
+bool setJointPosition(Waypoint& waypoint, const Eigen::Ref<const Eigen::VectorXd>& position)
+{
+  if (isJointWaypoint(waypoint))
+    *waypoint.cast<JointWaypoint>() = position;
+  else if (isStateWaypoint(waypoint))
+    waypoint.cast<StateWaypoint>()->position = position;
+  else
+    return false;
+
+  return true;
+}
+
+bool clampToJointLimits(Waypoint& wp, const Eigen::Ref<const Eigen::MatrixX2d>& limits)
```

Comment:
Should the clamping only be performed if within some tolerance?

Reply:
Yeah. Probably. I was being lazy. I'll fix it.